code/clue.js

```diff
@@ -1,59 +1,325 @@
-const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
-}
+const randomSelector = (array) => {
+  return array[Math.floor(Math.random() * array.length)];
+};
 
 // CREATE AN OBJECT THAT KEEPS THE MYSTERY.
 // With a killer, a weapon and a room.
 // The values will be set later.
+const mystery = {
+  killer: null,
+  weapon: null,
+  room: null,
+};
+
+// This function will randomly change favourite weapons of each person
+const shuffleFavouriteWeapon = () => {
+  suspects.forEach((suspect) => {
+    const randomWeapon = randomSelector(weapons);
+    suspect.favouriteWeapon = randomWeapon.id;
+  });
+};
+
+const cleanRevealMysteryMessage = () => {
+  document.getElementById("mystery").innerHTML = "";
+};
+
+const toggleAnimationDeckCard = (id) => {
+  document.getElementById(id).classList.toggle("deck-heartbeat");
+};
 
 // This function will be invoked when you click on the killer card.
-const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+const pickKiller = (event) => {
+  event.preventDefault();
+  if (!killerPicked) {
+    // This will remove the reveal the crime message if any
+    cleanRevealMysteryMessage();
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+    toggleAnimationDeckCard("killerCard");
+
+    setTimeout(() => {
+      shuffleFavouriteWeapon();
+
+      // This will randomly select a killer from the suspects. And add that to the mystery object.
+      mystery.killer = randomSelector(suspects);
+
+      // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
+      document.getElementById("killerCard").style.background =
+        mystery.killer.color;
+      document.getElementById(
+        "killerName"
+      ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+      document.getElementById(
+        "killerAge"
+      ).innerHTML = `Age: ${mystery.killer.age}`;
+      document.getElementById(
+        "killerOccupation"
+      ).innerHTML = `Occupation: ${mystery.killer.occupation}`;
+      document.getElementById("killerImage").src = `${mystery.killer.image}`;
+      document.getElementById(
+        "killerDescription"
+      ).innerHTML = `${mystery.killer.description}`;
+      document.getElementById(
+        "killerFavouriteWeapon"
+      ).innerHTML = `Favourite Weapon: ${mystery.killer.favouriteWeapon}`;
```

Comment:
I get the same formatting, it makes it so hard to read! If you find out how to change it please let me know